react-native-heap.podspec

```diff
@@ -3,6 +3,52 @@ require "json"
 json = File.read(File.join(__dir__, "package.json"))
 package = JSON.parse(json).deep_symbolize_keys
 
+HEAP_CONFIG_FILENAME = 'heap.config.json'
+
+# This function creates a `HeapSettings` plist file that the Obj-C library
+# uses for initialization. This is a bit of a hack, but I couldn't find a better way:
+#
+#    - Client apps couldn't directly access .xcconfig files from a static library.
+#    - Ditto for Info.plist files, which are stripped since this library is statically linked.
+#
+# Instead, simply declare a new "resource bundle" containing just a single plist file with the
+# settings, and load it from Obj-C.
+#
+# KLUDGE: Cocoapods loads this file in a way that doesn't allow direct method calls, so we use a Proc here instead.
+write_heap_settings = Proc.new do
+  # `pwd` is <client_app>/node_modules/@heap/react-native-heap, so we go up three levels.
+  settings_filename = File.expand_path("../../../#{HEAP_CONFIG_FILENAME}")
+
+  if File.file?(settings_filename)
+    json = JSON.parse(File.read(settings_filename).strip).deep_symbolize_keys
+    dev, prod, default = json.values_at(:dev, :prod, :default)
+
+    auto_init_dev = dev[:heapAutoInit]
+    auto_init_dev = default[:heapAutoInit] if auto_init_dev.nil?
+    auto_init_dev = true if auto_init_dev.nil?
+
+    auto_init_prod = prod[:heapAutoInit]
+    auto_init_prod = default[:heapAutoInit] if auto_init_prod.nil?
+    auto_init_prod = true if auto_init_prod.nil?
+
+    app_id_dev = dev[:heapAppId] || default[:heapAppId] if auto_init_dev
+    app_id_prod = prod[:heapAppId] || default[:heapAppId] if auto_init_prod
+  else
+    puts "Not auto-initializing Heap; couldn't find #{HEAP_CONFIG_FILENAME}."
+  end
+
+  template = File.read(File.join(__dir__, 'ios', 'HeapSettings', 'HeapSettings.plist.template'))
+  output_path = File.join(__dir__, 'ios', 'HeapSettings', 'HeapSettings.plist')
+
+  settings = template
```

Comment:
I would suggest using `plistbuddy` for something like this.  See, e.g., https://willowtreeapps.com/ideas/using-plistbuddy-to-inject-app-secrets-and-keys-at-build-time for a very similar use case.